core/src/main/scala/org/apache/spark/deploy/FaultToleranceTest.scala

```diff
@@ -279,7 +279,7 @@ private object FaultToleranceTest extends App with Logging {
     var liveWorkerIPs: Seq[String] = List()
 
     def stateValid(): Boolean = {
-      (workers.map(_.ip) -- liveWorkerIPs).isEmpty &&
+      workers.map(_.ip).forall(liveWorkerIPs.contains) &&
```

Comment:
Nit: What about using `diff` here?
As I see `diff` is not deprecated: https://www.scala-lang.org/api/current/scala/collection/Seq.html#diff[B%3E:A](that:scala.collection.Seq[B]):C

 ```suggestion
      workers.map(_.ip).diff(liveWorkerIPs).isEmpty &&
```

Reply:
diff would work too, I think. It has multiset semantics, and I thought it was not necessary here. I went for what I thought was simpler, but I am not 100% sure.